KML overlay, Hexagon Control, Performance settings tutorial completed

risehr opened the pull request:
Completed all above mentioned tutorials
Fixed bugs/ logical errors in screen recorder class
Restricted recording window size to cover proper display

risehr committed:
Kml, hexagon control, and performance settings tutorial completed

risehr committed:
Improved the screenrecorder class fixing some bugs, improved restricted size recording of windows

risehr committed:
Minor changes in screen recorder class

ReimarBauer commented on tutorials/tutorial_hexagoncontrol.py:
```diff
@@ -0,0 +1,386 @@
+"""
+    mss.tutorials.tutorial_hexagoncontrol.py
+    ~~~~~~~~~~~~~~~~~~~
```
the underline must have same length as the name

ReimarBauer commented on tutorials/tutorial_kml.py:
```diff
@@ -0,0 +1,340 @@
+"""
+    mss.tutorials.tutorial_kml
+    ~~~~~~~~~~~~~~~~~~~
```
line length, of `~~~` same length as the word above

risehr replied:
Okay. I will look into the lines length.

For the Exception part, the pyautogui.locateCentreOnScreen (line no 178) has screen coordinates given that which area to search and may be your table view 's select to open control position is not showing up in that region.
It needs to be from the middle to right of the screen I guess.
On my laptop it is working.

risehr replied:
Can you please try to change those coordinates a bit and run. I am figuring out a way that it doesn't show that exception. @ReimarBauer

ReimarBauer replied:
out of the box the tutorial_performancesettings.py is able to open the subwidget, while the same menu is not working with the tutorial_hexagoncontrol.py.

hexagoncontrol tutorial does not find on my system the locastion of selecttoopencontrol.png  and can't open that.

risehr replied:
This time, I have used selecttoopencontrol.png file to find the location of the table view window header and drag it to bottom. So if it is dragging, its found the selecttoopencontrol image file and acted accordingly. Next, I used a relative location that moves the cursor on it and clicks it opening the drop down menu of hexagon control. Don't understand why its not working. Have you tried with the updated commit?

ReimarBauer reviewed (changes requested):
I have a problem with

```
(risehr) reimar@infinte:~/PycharmProjects/2021/risehr/MSS/tutorials$ python ./tutorial_hexagoncontrol.py 


Exception :'select to open control(table view)' button/option not found on the screen.

Exception :'Center Latitude' button not found on the screen.

Exception :'Add Hexagon' button not found on the screen.

Exception :'Radius' button not found on the screen.

Exception :'Remove Hexagon' button not found on the screen.

Exception :'Add Hexagon' button not found on the screen.

Exception :'Radius (for Angle of first point)' button not found on the screen.

Automation is over for this tutorial. Watch next tutorial for other functions.
.
```
Please have a look on this

risehr committed:
made small changes in the header files of scipt and other minor changes in hexagon tutorial

risehr commented:
I have made the changes. Just see once, they work or not. @ReimarBauer

risehr committed:
Made the code more efficient by small changes in  hexagon control tutorial

risehr commented:
I have tried again. I don't know if it works in your system or not!

ReimarBauer commented:
I changed a bit the order and added some ideas how we can improve it at all

[risehr.patch.txt](https://github.com/Open-MSS/MSS/files/6986141/risehr.patch.txt)

ReimarBauer commented:
> I changed a bit the order and added some ideas how we can improve it at all
> 
> [risehr.patch.txt](https://github.com/Open-MSS/MSS/files/6986141/risehr.patch.txt)

There are some "q" on some places. That happens sometimes by the script running.

risehr commented:
> > I changed a bit the order and added some ideas how we can improve it at all
> > [risehr.patch.txt](https://github.com/Open-MSS/MSS/files/6986141/risehr.patch.txt)
> 
> There are some "q" on some places. That happens sometimes by the script running.

I guess, the same problem is with the screen recorder window. It does not come to the front by alt + tab + left. And some other window like your terminal appears later and then the "q" got pressed there.

In my system, the screen recorder comes to the front by the alt + tab + left and then pressing "q" closes it peacefully.

ReimarBauer commented:
> > > I changed a bit the order and added some ideas how we can improve it at all
> > > [risehr.patch.txt](https://github.com/Open-MSS/MSS/files/6986141/risehr.patch.txt)
> > 
> > 
> > There are some "q" on some places. That happens sometimes by the script running.
> 
> I guess, the same problem is with the screen recorder window. It does not come to the front by alt + tab + left. And some other window like your terminal appears later and then the "q" got pressed there.
> 
> In my system, the screen recorder comes to the front by the alt + tab + left and then pressing "q" closes it peacefully.

For the screen recorder we maybe could use  pag.locateCenterOnScreen with its icon in the task manager and click on it.

If I press Alt Down + Tab Down  and Tab Up (while Alt is still down) it looks like this

![alt_tab](https://user-images.githubusercontent.com/278728/129442345-d80c26f4-2378-492b-abdd-376321d64dbe.gif)

When I click by the mouse on an icon I select it and then by Alt Up this program is activated on screen. 

Perhaps when we use the icon for its location it may be also found in the task bar?

risehr commented:
> > > > I changed a bit the order and added some ideas how we can improve it at all
> > > > [risehr.patch.txt](https://github.com/Open-MSS/MSS/files/6986141/risehr.patch.txt)
> > > 
> > > 
> > > There are some "q" on some places. That happens sometimes by the script running.
> > 
> > 
> > I guess, the same problem is with the screen recorder window. It does not come to the front by alt + tab + left. And some other window like your terminal appears later and then the "q" got pressed there.
> > In my system, the screen recorder comes to the front by the alt + tab + left and then pressing "q" closes it peacefully.
> 
> For the screen recorder we maybe could use pag.locateCenterOnScreen with its icon in the task manager and click on it.
> 
> If I press Alt Down + Tab Down and Tab Up (while Alt is still down) it looks like this
> 
> ![alt_tab](https://user-images.githubusercontent.com/278728/129442345-d80c26f4-2378-492b-abdd-376321d64dbe.gif)
> 
> When I click by the mouse on an icon I select it and then by Alt Up this program is activated on screen.
> 
> Perhaps when we use the icon for its location it may be also found in the task bar?

Yes, we can use that icon that way. But since it has an icon of terminal, other things may also be clicked that are opened with terminal.

ReimarBauer commented:
> > > > > I changed a bit the order and added some ideas how we can improve it at all
> > > > > [risehr.patch.txt](https://github.com/Open-MSS/MSS/files/6986141/risehr.patch.txt)
> > > > 
> > > > 
> > > > There are some "q" on some places. That happens sometimes by the script running.
> > > 
> > > 
> > > I guess, the same problem is with the screen recorder window. It does not come to the front by alt + tab + left. And some other window like your terminal appears later and then the "q" got pressed there.
> > > In my system, the screen recorder comes to the front by the alt + tab + left and then pressing "q" closes it peacefully.
> > 
> > 
> > For the screen recorder we maybe could use pag.locateCenterOnScreen with its icon in the task manager and click on it.
> > If I press Alt Down + Tab Down and Tab Up (while Alt is still down) it looks like this
> > ![alt_tab](https://user-images.githubusercontent.com/278728/129442345-d80c26f4-2378-492b-abdd-376321d64dbe.gif)
> > When I click by the mouse on an icon I select it and then by Alt Up this program is activated on screen.
> > Perhaps when we use the icon for its location it may be also found in the task bar?
> 
> Yes, we can use that icon that way. But since it has an icon of terminal, other things may also be clicked that are opened with terminal.

found this idea, may be you can enable it just at the end and move the mouse cursor to that mentioned position?
https://stackoverflow.com/questions/65068775/how-to-stop-or-pause-pyautogui-at-any-moment-that-i-want

ReimarBauer commented:
> > > > > I changed a bit the order and added some ideas how we can improve it at all
> > > > > [risehr.patch.txt](https://github.com/Open-MSS/MSS/files/6986141/risehr.patch.txt)
> > > > 
> > > > 
> > > > There are some "q" on some places. That happens sometimes by the script running.
> > > 
> > > 
> > > I guess, the same problem is with the screen recorder window. It does not come to the front by alt + tab + left. And some other window like your terminal appears later and then the "q" got pressed there.
> > > In my system, the screen recorder comes to the front by the alt + tab + left and then pressing "q" closes it peacefully.
> > 
> > 
> > For the screen recorder we maybe could use pag.locateCenterOnScreen with its icon in the task manager and click on it.
> > If I press Alt Down + Tab Down and Tab Up (while Alt is still down) it looks like this
> > ![alt_tab](https://user-images.githubusercontent.com/278728/129442345-d80c26f4-2378-492b-abdd-376321d64dbe.gif)
> > When I click by the mouse on an icon I select it and then by Alt Up this program is activated on screen.
> > Perhaps when we use the icon for its location it may be also found in the task bar?
> 
> Yes, we can use that icon that way. But since it has an icon of terminal, other things may also be clicked that are opened with terminal.

You may be can add to the screen recorder something similiar as pyautogui has, described here
https://stackoverflow.com/questions/65068775/how-to-stop-or-pause-pyautogui-at-any-moment-that-i-want

so that moving the mouse into the screens corner will terminate it.

ReimarBauer reviewed (approved):
discussed to add missing parts in another PR
